agent/obfuscate.js

```diff
@@ -25,20 +25,27 @@ function getRules () {
 function validateRules (rules) {
   var invalidReplacementDetected = false
   var invalidRegexDetected = false
-
   for (var i = 0; i < rules.length; i++) {
-    if (!('regex' in rules[i]) || typeof rules[i].regex !== 'string' || !(rules.regex instanceof RegExp)) {
-      if (console && console.warn) console.warn('Regex: ', rules[i].regex, ' with replacement ', rules[i].replacement, 'invalid')
+    if (!('regex' in rules[i])) {
```

Comment:
I noticed you removed the check for regex here. I saw a minor problem in the original snippet that was removed that gets fixed if you do.  We can discuss if this is something we need

```if (!('regex' in rules[i]) || !(typeof rules[i].regex !== 'string' || !(rules.regex instanceof RegExp)))````

Reply:
After discussion, we're keeping separate warnings for missing regex arg and invalid regex arg